### PR TITLE
pull out static objects to factory for package_hooks_linter

### PR DESCRIPTION
**Pull request by MichaelChirico**
Part of #1199 

Here's a case where deprecating `is.function(lint_message)` will take a bit more effort -- currently the function is nested a few layers deep, vectorizing is not quite as trivial.

### CODE REVIEW

File: R/package_hooks_linter.R [hunk 1]
```diff
@@ -15,141 +15,133 @@
...
+    ".onAttach" = c(bad_msg_calls, "library.dynam")
+  )
+
+  # lints here will hit the function <expr>,
+  #   this path returns to the corresponding namespace hook's name
+  ns_calls <- xp_text_in_table(c(".onLoad", ".onAttach", ".onDetach", ".Last.lib"))
+  hook_xpath <- sprintf("string(./ancestor::expr/expr/SYMBOL[%s])", ns_calls)
+  get_hook <- function(xml) {
+    xml2::xml_find_chr(xml, hook_xpath)
+  }
```
Comment (AshesITR) [inline R/package_hooks_linter.R:30]: Is the object_usage_lint correct?

Comment (MichaelChirico) [reply R/package_hooks_linter.R:30]: yep! forgot to delete this. good bot

**Review by AshesITR — DISMISSED** [review]

File: R/package_hooks_linter.R [hunk 2]
```diff
@@ -15,141 +15,130 @@
...
-        } else {
-          sprintf("Don't alter the search() path in %s() by calling %s().", hook, bad_call)
-        }
-      },
-      type = "warning"
-    )
+    library_require_bad_call <- xml2::xml_text(library_require_expr)
+    library_require_hook <- xml2::xml_find_chr(library_require_expr, hook_xpath)
+    library_require_message <- character(length(library_require_bad_call))
+    is_install_packages <- library_require_bad_call == "installed.packages"
```
Comment (AshesITR) [inline R/package_hooks_linter.R:115]: typo -> `is_installed_packages`

**Comment by AshesITR** [comment]
Still not too bad imo

**Review by AshesITR — APPROVED** [review]

**Review by AshesITR — APPROVED** [review]

**Pull request merged**